app/src/main/java/samyups/example/trainingss/MainActivity.kt

```diff
@@ -1,11 +1,28 @@
 package samyups.example.trainingss
 
-import android.support.v7.app.AppCompatActivity
+import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
+
+    val dogAdapter = DogAdapter()
+    private var layoutManager: RecyclerView.LayoutManager? = null
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        initDogRV()
+    }
+
+    private fun initDogRV() {
```

Comment:
This method name is not good. RV is pretty generic. In this case I know it means RecyclerView, but usually you want to make method/variable names crystal clear. I would call this method `initRecyclerView`. We want to leave the `Dog` out of it because you might change the type of data in here at some point, and then you have t go and change all these method names to reflect that. And if you forget to change the method names, you just have mis-named stuff everywhere.